code/modules/jobs/job_types/job.dm

```diff
@@ -81,12 +81,10 @@
 	if(!H)
 		return FALSE
 	if(!visualsOnly)
-		var/obj/item/card/id/I = H.wear_id ? H.wear_id.GetID() : null
-		var/datum/bank_account/bank_account
-		if(I)
-			bank_account = new(I.registered_name, src)
-		else
-			bank_account = new(H.real_name, src)
+		var/datum/bank_account/bank_account = new(H.real_name, src)
+		bank_account.account_holder = H.real_name
+		bank_account.account_job = src
+		bank_account.account_id = rand(111111,999999)

```

Comment:
A little nitpicky I know but I think these three lines don't need to be here, if they're being handled by `/datum/bank_account/New()` already